src/node_errors.h

```diff
@@ -132,6 +132,22 @@ inline v8::Local<v8::Value> ERR_STRING_TOO_LONG(v8::Isolate* isolate) {
                                               prefix " must be a string");   \
   } while (0)
 
+namespace errors {
+
+class TryCatch : public v8::TryCatch {
+ public:
+  enum { kNormal, kFatal };
+
+  explicit TryCatch(Environment* env, int mode = kNormal)
+      : v8::TryCatch(env->isolate()), env_(env), mode_(mode) {}
+  ~TryCatch();
```

Comment:
This is a little problematic, since `v8::TryCatch::~TryCatch` isn't virtual, we need to make this class isn't polymorphic. One possibility is to replace the `is a` abstraction (i.e. public inheritance) to `has a` (i.e. `v8::TryCatch` as a private member).

Reply:
Can you explain a bit more what the problem is here? This bit of code has been working fine for almost a year.

Reply:
FTR: @devsnek and I talked on [IRC](http://logs.libuv.org/node-dev/2018-10-27#20:50:03.437), and came to the conclusion it would be better to simply delete `operator new()`.
```suggestion
  ~TryCatch();

  // Since the dtor is not virtual we need to make sure no one creates object of in the free store that might be held by polymorphic pointers.
  // Please please please noone do `v8::TryCatch& = node::TryCatch();`
  void* operator new  ( std::size_t count ) = delete;
  void* operator new[]( std::size_t count ) = delete;
```